plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2954,23 +2967,29 @@ private void deleteUnregisteredVmFiles(VirtualMachineFileLayoutEx vmFileLayout,
                 else if (file.getType().equals("config"))
                     vmFolder = new DatastoreFile(fileInDatastore.getDatastoreName(), fileInDatastore.getDir());
                 DatastoreMO dsMo = new DatastoreMO(dcMo.getContext(), dcMo.findDatastore(fileInDatastore.getDatastoreName()));
-                s_logger.debug("Deleting file: " + file.getName());
-                dsMo.deleteFile(file.getName(), dcMo.getMor(), true, VmwareManager.s_vmwareSearchExcludeFolder.value());
+                if (skipDatastores == null || !skipDatastores.contains(dsMo.getName())) {
+                    s_logger.debug("Deleting file: " + file.getName());
+                    dsMo.deleteFile(file.getName(), dcMo.getMor(), true);
+                }
             }
             // Delete files that are present in the VM folder - this will take care of the VM disks as well.
             DatastoreMO vmFolderDsMo = new DatastoreMO(dcMo.getContext(), dcMo.findDatastore(vmFolder.getDatastoreName()));
             String[] files = vmFolderDsMo.listDirContent(vmFolder.getPath());
             if (deleteDisks) {
                 for (String file : files) {
                     String vmDiskFileFullPath = String.format("%s/%s", vmFolder.getPath(), file);
-                    s_logger.debug("Deleting file: " + vmDiskFileFullPath);
-                    vmFolderDsMo.deleteFile(vmDiskFileFullPath, dcMo.getMor(), true, VmwareManager.s_vmwareSearchExcludeFolder.value());
+                    if (skipDatastores == null || !skipDatastores.contains(vmFolderDsMo.getName())) {
+                        s_logger.debug("Deleting file: " + vmDiskFileFullPath);
+                        vmFolderDsMo.deleteFile(vmDiskFileFullPath, dcMo.getMor(), true);
+                    }
                 }
             }
             // Delete VM folder
             if (deleteDisks || files.length == 0) {
-                s_logger.debug("Deleting folder: " + vmFolder.getPath());
-                vmFolderDsMo.deleteFolder(vmFolder.getPath(), dcMo.getMor());
+                if (skipDatastores == null || !skipDatastores.contains(vmFolderDsMo.getName())) {
```

Comment:
there is some repetition here. maybe worth extracting into a method. (even though two are for files and one for a folder)

Reply:
Will check and update. thanks for the review @DaanHoogland